lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -12,357 +12,365 @@
 require 'fluent/output'
 require_relative 'elasticsearch_index_template'
 
-class Fluent::ElasticsearchOutput < Fluent::ObjectBufferedOutput
-  class ConnectionFailure < StandardError; end
-
-  Fluent::Plugin.register_output('elasticsearch', self)
-
-  config_param :host, :string,  :default => 'localhost'
-  config_param :port, :integer, :default => 9200
-  config_param :user, :string, :default => nil
-  config_param :password, :string, :default => nil, :secret => true
-  config_param :path, :string, :default => nil
-  config_param :scheme, :string, :default => 'http'
-  config_param :hosts, :string, :default => nil
-  config_param :target_index_key, :string, :default => nil
-  config_param :target_type_key, :string, :default => nil
-  config_param :time_key_format, :string, :default => nil
-  config_param :logstash_format, :bool, :default => false
-  config_param :logstash_prefix, :string, :default => "logstash"
-  config_param :logstash_dateformat, :string, :default => "%Y.%m.%d"
-  config_param :utc_index, :bool, :default => true
-  config_param :type_name, :string, :default => "fluentd"
-  config_param :index_name, :string, :default => "fluentd"
-  config_param :id_key, :string, :default => nil
-  config_param :write_operation, :string, :default => "index"
-  config_param :parent_key, :string, :default => nil
-  config_param :routing_key, :string, :default => nil
-  config_param :request_timeout, :time, :default => 5
-  config_param :reload_connections, :bool, :default => true
-  config_param :reload_on_failure, :bool, :default => false
-  config_param :resurrect_after, :time, :default => 60
-  config_param :time_key, :string, :default => nil
-  config_param :time_key_exclude_timestamp, :bool, :default => false
-  config_param :ssl_verify , :bool, :default => true
-  config_param :client_key, :string, :default => nil
-  config_param :client_cert, :string, :default => nil
-  config_param :client_key_pass, :string, :default => nil
-  config_param :ca_file, :string, :default => nil
-  config_param :remove_keys, :string, :default => nil
-  config_param :remove_keys_on_update, :string, :default => ""
-  config_param :remove_keys_on_update_key, :string, :default => nil
-  config_param :flatten_hashes, :bool, :default => false
-  config_param :flatten_hashes_separator, :string, :default => "_"
-  config_param :template_name, :string, :default => nil
-  config_param :template_file, :string, :default => nil
-  config_param :templates, :hash, :default => nil
-  config_param :include_tag_key, :bool, :default => false
-  config_param :tag_key, :string, :default => 'tag'
-  config_param :time_parse_error_tag, :string, :default => 'Fluent::ElasticsearchOutput::TimeParser.error'
-  config_param :reconnect_on_error, :bool, :default => false
-
-  include Fluent::ElasticsearchIndexTemplate
-
-  def initialize
-    super
-  end
+module Fluent::Plugin
+  class Fluent::ElasticsearchOutput < Output
+    class ConnectionFailure < StandardError; end
+
```

Comment:
This is because original `router` definition is not used in test.
If we want to test this case, it needs to adjust  for method definitions with `stub` or `mock` from `test-unit-rr`.